var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -781,36 +651,49 @@ def define_tpl_enable(cmake_var, spec_var=None):
                     define('Tpetra_INST_INT_LONG_LONG', gotype == 'long_long'),
                 ])
 
-        # disable due to compiler / config errors:
-        if spec.satisfies('%xl') or spec.satisfies('%xl_r'):
+        # ################# Kokkos ######################
+
+        if '+kokkos' in spec:
+            arch = Kokkos.spack_micro_arch_map.get(spec.target.name, None)
+            if arch:
+                options.append(define("Kokkos_ARCH_" + arch.upper(), True))
+
+            define_kok_enable = _make_definer("Kokkos_ENABLE_")
             options.extend([
-                define('Trilinos_ENABLE_Pamgen', False),
-                define('Trilinos_ENABLE_Stokhos', False),
+                define_kok_enable('CUDA'),
+                define_kok_enable('OPENMP' if spec.version >= Version('13')
+                                  else 'OpenMP'),
             ])
+            if '+cuda' in spec:
+                options.extend([
+                    define_kok_enable('CUDA_UVM', True),
+                    define_kok_enable('CUDA_LAMBDA', True),
+                    define_kok_enable('CUDA_RELOCATABLE_DEVICE_CODE', 'cuda_rdc')
+                ])
+                arch_map = Kokkos.spack_cuda_arch_map
+                options.extend(
+                    define("Kokkos_ARCH_" + arch_map[arch].upper(), True)
+                    for arch in spec.variants['cuda_arch'].value
+                )
 
-        if sys.platform == 'darwin':
-            options.append(define('Trilinos_ENABLE_FEI', False))
-            if '+stk' in spec:
-                cxx_flags.extend(['-DSTK_NO_BOOST_STACKTRACE'])
+        # ################# System-specific ######################
+
+        # Fortran lib (assumes clang is built with gfortran!)
+        if ('+fortran +mpi' in spec
+                and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
+            mpifc = Executable(spec['mpi'].mpifc)
+            libgfortran = mpifc('--print-file-name libgfortran.a', output=str)
```

Comment:
Gitlab is failing building trilinos, and it seems the error is related to the `--print-file-name` option

Reply:
Aha, it's because I didn't update the shell-like parsing before to use the separate arguments of a spack Executable. Thanks.